packages/block-library/src/footnotes/index.php

```diff
@@ -83,136 +83,130 @@ function register_block_core_footnotes() {
 }
 add_action( 'init', 'register_block_core_footnotes' );
 
-add_action(
-	'wp_after_insert_post',
-	/**
-	 * Saves the footnotes meta value to the revision.
-	 *
-	 * @since 6.3.0
-	 *
-	 * @param int $revision_id The revision ID.
-	 */
-	static function( $revision_id ) {
-		$post_id = wp_is_post_revision( $revision_id );
-
-		if ( $post_id ) {
+/**
+ * Saves the footnotes meta value to the revision.
+ *
+ * @since 6.3.0
+ *
+ * @param int $revision_id The revision ID.
+ */
+function wp_save_footnotes_meta( $revision_id ) {
+	$post_id = wp_is_post_revision( $revision_id );
+
+	if ( $post_id ) {
+		$footnotes = get_post_meta( $post_id, 'footnotes', true );
+
+		if ( $footnotes ) {
+			// Can't use update_post_meta() because it doesn't allow revisions.
+			update_metadata( 'post', $revision_id, 'footnotes', $footnotes );
+		}
+	}
+}
+add_action( 'wp_after_insert_post', 'wp_save_footnotes_meta' );
+
+/**
+ * Keeps track of the revision ID for "rest_after_insert_{$post_type}".
+ *
+ * @since 6.3.0
+ *
+ * @param int $revision_id The revision ID.
+ */
+function wp_keep_revision_id( $revision_id ) {
+	global $wp_temporary_footnote_revision_id;
+	$wp_temporary_footnote_revision_id = $revision_id;
+}
+add_action( '_wp_put_post_revision', 'wp_keep_revision_id' );
+
+/**
+ * This is a specific fix for the REST API. The REST API doesn't update
+ * the post and post meta in one go (through `meta_input`). While it
+ * does fix the `wp_after_insert_post` hook to be called correctly after
+ * updating meta, it does NOT fix hooks such as post_updated and
+ * save_post, which are normally also fired after post meta is updated
+ * in `wp_insert_post()`. Unfortunately, `wp_save_post_revision` is
+ * added to the `post_updated` action, which means the meta is not
+ * available at the time, so we have to add it afterwards through the
+ * `"rest_after_insert_{$post_type}"` action.
+ *
+ * @since 6.3.0
+ *
+ * @global int $wp_temporary_footnote_revision_id The footnote revision ID.
+ *
+ * @param WP_Post $post The post object.
+ */
+function wp_add_revisions_to_post_meta( $post ) {
```

Comment:
Should this be more specific to reflect functionality? E.g., `wp_add_footnotes_revisions_to_post_meta`